code/defines/procs/announce.dm

```diff
@@ -1,83 +1,79 @@
-GLOBAL_DATUM_INIT(minor_announcement, /datum/announcement/minor, new())
-GLOBAL_DATUM_INIT(priority_announcement, /datum/announcement/priority, new(do_log = FALSE))
-GLOBAL_DATUM_INIT(command_announcement, /datum/announcement/priority/command, new(do_log = FALSE))
-GLOBAL_DATUM_INIT(event_announcement, /datum/announcement/priority/command/event, new(do_log = FALSE))
-
-/datum/announcement
-	var/title = "Attention"
-	var/announcer = ""
-	var/log = FALSE
-	var/sound
-	var/channel_name = "Station Announcements"
-	var/announcement_type = "Announcement"
-	var/admin_announcement = FALSE // Admin announcements are received regardless of being in range of a radio, unless you're in the lobby to prevent metagaming
+GLOBAL_DATUM_INIT(minor_announcement, /datum/announcer, new(config_type = /datum/announcement_configuration/minor))
+GLOBAL_DATUM_INIT(major_announcement, /datum/announcer, new(config_type = /datum/announcement_configuration/major))
+GLOBAL_DATUM_INIT(event_announcement, /datum/announcer, new(config_type = /datum/announcement_configuration/event))
+
+/datum/announcement_configuration
+	var/default_title = "Attention"
+	/// The name used when describing the announcement type in logs.
+	var/log_name = ANNOUNCE_KIND_DEFAULT
+	/// Whether or not to log the announcement when made.
+	var/add_log = TRUE
```

Comment:
I leaned towards thinking we'd want logging on by default to prevent oversights where e.g. another announcer is used by players that doesn't get logged. The actual config overall gets a lot cleaner with your suggestion and I'm probably prematurely future-proofing :-)